app/lib/backend_version.rb

```diff
@@ -20,6 +20,10 @@ def oidc_usable?(_)
       false
     end
 
+    def oauth_usable?(service)
+      service.provider.provider_can_use?(:oauth_api) || service.oidc?
```

Comment:
I think it should be `&&`

Reply:
not really. then user would be able to set `oauth` only through `oidc` option.

this way, saas users will be able to set simple `oauth`.

`oauth` will be usable for onprem only if `oidc` option has been picked.

Reply:
But we do not want oauth at all in the on-prem version

Reply:
![image](https://user-images.githubusercontent.com/64276/49073644-35859180-f233-11e8-8458-ba832149e586.png)

Reply:
Refactoring this functionality is complex so will be done later.